string_calculator.rb

```diff
@@ -0,0 +1,42 @@
+class StringCalculator
+  def add(string_of_numbers = "")
+    sum = 0
+    separate_string = string_of_numbers.split(",")
+    numbers = separate_string.collect do |n|
+      n.to_i
+    end
+    neg = numbers.select(&:negative?)
+    is_negative = numbers.any? { |x| x.negative? }
```

Comment:
If there is a negative number, you can just return early. You don't have to wait until line 22.

```ruby
return "negatives not allowed: #{neg.to_s}" if numbers.any?(&:negative?)
```

You can also use the shorthand by passing the symbol `:negative?` to `any?`.

Reply:
Done, thank you for the tip